File: app/src/main/java/net/kibotu/android/recyclerviewpresenter/app/java/LabelPresenter.java
package net.kibotu.android.recyclerviewpresenter.app.java;

import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import net.kibotu.android.recyclerviewpresenter.Presenter;
import net.kibotu.android.recyclerviewpresenter.PresenterViewModel;
import net.kibotu.android.recyclerviewpresenter.app.R;
import net.kibotu.android.recyclerviewpresenter.app.databinding.LabelPresenterItemBinding;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

/**
 * Created by <a href="https://kibotu.net">Jan Rabe</a>.
 */

public class LabelPresenter extends Presenter<String, LabelPresenterItemBinding> {

    public LabelPresenter() {
        super(R.layout.label_presenter_item, LabelPresenterItemBinding::bind);
    }

    @Override
    public void bindViewHolder(@NonNull LabelPresenterItemBinding viewBinding, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull PresenterViewModel<String> item, @androidx.annotation.Nullable List<Object> payloads) {
        viewBinding.label.setText(item.getModel());
    }
}

File: app/src/main/java/net/kibotu/android/recyclerviewpresenter/app/kotlin/LabelPresenter.kt
package net.kibotu.android.recyclerviewpresenter.app.kotlin

import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.label_presenter_item.view.*
import net.kibotu.android.recyclerviewpresenter.Adapter
import net.kibotu.android.recyclerviewpresenter.Presenter
import net.kibotu.android.recyclerviewpresenter.PresenterModel
import net.kibotu.android.recyclerviewpresenter.app.R
import net.kibotu.logger.Logger.logv
import net.kibotu.logger.logv

/**
 * Created by [Jan Rabe](https://about.me/janrabe).
 */
class LabelPresenter : Presenter<String>() {

    override val layout = R.layout.label_presenter_item

    override fun bindViewHolder(viewHolder: RecyclerView.ViewHolder, item: PresenterModel<String>, position: Int, payloads: MutableList<Any>?, adapter: Adapter) = with(viewHolder.itemView) {
        logv("bindViewHolder $position $item payload=$payloads")
        label.text = "$position. ${item.model}"
    }
}